### PR TITLE
feat(service-portal): Settings options for paper mail requirement

### PR DESCRIPTION
**Pull request by thordurhhh**
## What

* Add settings options for paper mail requirement

NOTE: We have a big change coming up for the document service in the near future. So this PR will only add to the service as is. A re-write will be done in a few weeks.

## Why

New feature requirement

Screenshot 2023-03-14 at 14 28 20

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/documents/src/lib/document.service.ts [hunk 1]
```diff
@@ -154,4 +156,43 @@ export class DocumentService {
...
+
+  async getPaperMailInfo(nationalId: string): Promise<PaperMailBody> {
+    try {
+      const res = await this.documentClient.requestPaperMail(nationalId)
+      return {
+        nationalId: res?.kennitala,
+        wantsPaper: res?.wantsPaper,
+      }
+    } catch (exception) {
+      logger.error(exception)
```
Comment (thorkellmani) [inline libs/api/domains/documents/src/lib/document.service.ts:168]: Is the error message safe for direct logging? No sensitive stuff? Don't we want to add a LOG_CATEGORY or something?

File: libs/api/domains/documents/src/lib/document.service.ts [hunk 2]
```diff
@@ -154,4 +156,43 @@ export class DocumentService {
...
+        kennitala: nationalId,
+        wantsPaper: body.wantsPaper,
+      }
+      const res = await this.documentClient.postPaperMail(postBody)
+      return {
+        nationalId: res?.kennitala,
+        wantsPaper: res?.wantsPaper,
+      }
+    } catch (exception) {
+      logger.error(exception)
```
Comment (thorkellmani) [inline libs/api/domains/documents/src/lib/document.service.ts:191]: see above

File: libs/clients/documents/src/lib/documentClient.ts [hunk 3]
```diff
@@ -80,6 +82,35 @@ export class DocumentClient {
...
+        ),
+      )
+
+      return response.data
+    } catch (e) {
+      const errMsg = 'Failed to POST to Postholf'
+      const error = e.toJSON()
+      const description = error.message
+      const message = [errMsg, error, description].filter(Boolean).join(' - ')
+      throw new Error(message)
```
Comment (thorkellmani) [inline libs/clients/documents/src/lib/documentClient.ts:110]: this feels weird

File: libs/service-portal/settings/personal-information/src/components/Forms/ProfileForm/components/Inputs/PaperMail.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      await postPaperMailMutation({
+        variables: {
+          input: {
+            wantsPaper: data.wantsPaper,
+          },
+        },
+      })
+    } catch (err) {
+      console.error(`updateOrCreateUserProfile error: ${err}`)
+      setSubmitError(formatMessage(m.somethingWrong))
```
Comment (thorkellmani) [inline libs/service-portal/settings/personal-information/src/components/Forms/ProfileForm/components/Inputs/PaperMail.tsx:66]: No log?

**Review by thorkellmani — APPROVED** [review]
some questions

File: libs/service-portal/settings/personal-information/src/components/Forms/ProfileForm/components/Inputs/PaperMail.tsx (hunk 4 above)
Comment (thordurhhh) [reply libs/service-portal/settings/personal-information/src/components/Forms/ProfileForm/components/Inputs/PaperMail.tsx:66]: Error logged in the service, this is just to display error for the user.

File: libs/api/domains/documents/src/lib/document.service.ts (hunk 2 above)
Comment (thordurhhh) [reply libs/api/domains/documents/src/lib/document.service.ts:191]: This is just a continuation of the previous document service functions. Will re-visit on re-write

File: libs/api/domains/documents/src/lib/document.service.ts (hunk 1 above)
Comment (thordurhhh) [reply libs/api/domains/documents/src/lib/document.service.ts:168]: https://github.com/island-is/island.is/pull/10489#discussion_r1182608453

**Pull request merged**